rest/class-boldgrid-backup-rest-setting.php

```diff
@@ -0,0 +1,195 @@
+				'permission_callback' => [ $this, 'permission_check' ],
+			],
+			'schema' => [ $this, 'get_schema' ],
+		] );
+	}
+
+	/**
+	 * Register router for updating settings.
+	 *
+	 * @since X.X.X
+	 */
+	public function register_update() {
+		register_rest_route( $this->namespace, '/' . $this->resource, [
+			[
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => [ $this, 'update_item' ],
+				'permission_callback' => [ $this, 'permission_check' ],
+			],
+			'schema' => [ $this, 'get_schema' ],
+		] );
+	}
+
+	/**
+	 * Prepare the item for the REST response.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param mixed $item WordPress representation of the item.
+	 * @param WP_REST_Request $request Request object.
+	 * @return mixed
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		return $this->filter_schema_properties( $item );
+	}
+
+	/**
+	 * Get schema for settings.
+	 *
+	 * @since X.X.X
+	 *
+	 * @return array Schema Format.
+	 */
+	public function get_schema() {
+		$schema = [
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => $this->resource,
+			'type'       => 'object',
+			'properties' => [
+				'schedule'                 => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Date the backup was created.', 'boldgrid-backup' ),
+					'type'        => 'array',
+				],
+				'autoupdate'               => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Automatic Update.', 'boldgrid-backup' ),
+					'type'        => 'array',
+				],
+				'notification_email'       => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Email to notify got backups.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+				'auto_backup'              => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Does a site have auto backups enabled?', 'boldgrid-backup' ),
+					'type'        => 'integer',
+				],
+				'auto_backup'              => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Does a site have auto rollback enabled?', 'boldgrid-backup' ),
+					'type'        => 'integer',
+				],
+				'folder_exclusion_include' => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Directories and files to include.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+				'folder_exclusion_exclude' => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Directories and files to exclude.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+			],
+		];
+
+		return $schema;
+	}
+
+	/**
+	 * Get the users plugin settings.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return array                   Plugin settings.
+	 */
+	public function get_item( $request ) {
+		$settings          = get_option( 'boldgrid_backup_settings', [] );
```

Comment:
Arrays should be declared with long array syntax per WordPress coding standards
https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#declaring-arrays

Reply:
See similar comment above.